src/GraphQL/Queries/EntryQuery.php

```diff
@@ -53,6 +59,33 @@ public function resolve($root, $args)
             $query->where('site', $site);
         }
 
+        $this->filterQuery($query, $args['filter'] ?? []);
+
         return $query->limit(1)->get()->first();
     }
+
+    private function filterQuery($query, $filters)
+    {
+        if (! isset($filters['status']) && ! isset($filters['published'])) {
+            $filters['status'] = 'published';
+        }
+
+        foreach ($filters as $field => $definitions) {
+            if (! is_array($definitions)) {
+                $definitions = [['equals' => $definitions]];
+            }
+
+            if (Arr::assoc($definitions)) {
+                $definitions = collect($definitions)->map(function ($value, $key) {
+                    return [$key => $value];
+                })->values()->all();
+            }
+
+            foreach ($definitions as $definition) {
+                $condition = array_keys($definition)[0];
+                $value = array_values($definition)[0];
+                $this->queryCondition($query, $field, $condition, $value);
+            }
+        }
+    }
```

Comment:
Since this looks to be identical to what's in the `EntriesQuery` class, you could move this method into a trait and share it between both queries.

Reply:
Will do that, thanks!